pkg/roachpb/metadata.proto

```diff
@@ -318,11 +318,6 @@ message StoreCapacity {
   // by ranges in the store. The stat is tracked over the time period defined
   // in storage/replica_stats.go, which as of July 2018 is 30 minutes.
   optional double writes_per_second = 5 [(gogoproto.nullable) = false];
-  // l0_sublevels tracks the current number of l0 sublevels in the store.
-  // TODO(kvoli): Use of this field will need to be version-gated, to avoid
-  // instances where overlapping node-binary versions within a cluster result
-  // in this this field missing.
-  optional int64 l0_sublevels = 12 [(gogoproto.nullable) = false];
```

Comment:
In 23.1-22.2 mixed version clusters it will mean that 23.1 high l0 sublevel stores will still be rebalanced towards from 22.2 nodes.

It would be nice to just take the risk and remove it entirely, since this only prevents rebalancing by default. I've added the field back in.